lib/internal/url.js

```diff
@@ -201,7 +219,8 @@ class URLSearchParams {
           if ((typeof pair !== 'object' && typeof pair !== 'function') ||
               pair === null ||
               typeof pair[SymbolIterator] !== 'function') {
-            throw new ERR_INVALID_TUPLE('Each query pair', '[name, value]');
+            throwTypeError(
+              new ERR_INVALID_TUPLE('Each query pair', '[name, value]'));
```

Comment:
`ERR_INVALID_TUPLE` is already a `TypeError`

Reply:
It needs error.constructor equals to TypeError. So inheritance is not allowed.

Reply:
refs: https://github.com/nodejs/node/blob/master/test/fixtures/wpt/resources/testharness.js#L1662

Reply:
I would argue that's a bug in the test harness and not our implementation.

```js
const e = new ERR_INVALID_TUPLE('a', 'b');
console.log(e instanceof TypeError); // true
```

The spec text says only to throw a `TypeError`, and that is what we're doing here.

Reply:
our internal errors already satisfy this constraint:

```
> try { new URLSearchParams([1]) } catch(err) { console.log(err.constructor === TypeError) }
true
```

Reply:
2 ways to resolve it:

1. Ignore this case: new URLSearchParams(DOMException.prototype);
2. Fix the code to pass wpt.

Current code can't pass that case.

Reply:
It was fixed in https://github.com/nodejs/node/pull/33857

Reply:
I'll take a look later.

Reply:
But new URLSearchParams(DOMException.prototype); is thrown by DOMException.

Reply:
@targos You're right. The only thing I should fix for error handling is the `ERR_INVALID_THIS` in `lib/internal/per_context/domexception.js`.